modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Plugin/rest/resource/GatedContentCustomAuthentication.php

```diff
@@ -167,21 +183,56 @@ public function post(Request $request) {
   }
 
   /**
-   * Error Response.
-   *
-   * @param string $message
-   *   The error message.
-   * @param int $status
-   *   The HTTP status code for error.
-   *
-   * @return \Drupal\rest\ModifiedResourceResponse
-   *   The HTTP response object.
+   * Helper function for reCaptcha validation.
    */
-  protected function errorResponse($message, $status) {
+  protected function validateRecaptcha($content, $request) {
+    if (!$content['recaptchaToken']) {
+      return $this->errorResponse($this->t('ReCaptcha token required.'), 400);
+    }
+
+    $config = $this->configFactory->get('recaptcha.settings');
+    $recaptcha_secret_key = $config->get('secret_key');
+    $recaptcha = new ReCaptcha($recaptcha_secret_key, new Drupal8Post());
```

Comment:
looks like DI opportunity, can we?

Reply:
there no any services in reCaptcha module, this code was reused from `recaptcha_captcha_validation`

```php
/**
 * CAPTCHA Callback; Validates the reCAPTCHA code.
 */
function recaptcha_captcha_validation($solution, $response, $element, $form_state) {
  $config = \Drupal::config('recaptcha.settings');
  $recaptcha_secret_key = $config->get('secret_key');
  if (empty($_POST['g-recaptcha-response']) || empty($recaptcha_secret_key)) {
    return FALSE;
  }
  $recaptcha = new ReCaptcha($recaptcha_secret_key, new Drupal8Post());
  if ($config->get('verify_hostname')) {
    $recaptcha->setExpectedHostname($_SERVER['SERVER_NAME']);
  }
```

Maybe we can improve this later, I would prefer to leave it as is if it's not very urgent.